src/internal/analytics/__tests__/components/analytics-funnel.test.tsx

```diff
@@ -666,6 +667,53 @@ describe('AnalyticsFunnelSubStep', () => {
   });
 });
 
+describe('emit cloudscape-creation-flow-done event', () => {
+  const addCreateFlowDoneEvent = (fn: jest.Mock<any, any>) => {
+    document.addEventListener(CREATION_FLOW_DONE_EVENT_NAME, fn);
+  };
+  const removeCreateFlowDoneEvent = (fn: jest.Mock<any, any>) => {
+    document.removeEventListener(CREATION_FLOW_DONE_EVENT_NAME, fn);
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+    mockFunnelMetrics();
+  });
+
+  test('emit cloudscape-creation-flow-done event when funnelComplete', async () => {
+    const onCompleteSpy = jest.fn();
+    addCreateFlowDoneEvent(onCompleteSpy);
+    const ChildComponent = () => {
+      const { funnelSubmit } = useFunnel();
+      return <button onClick={funnelSubmit}>Submit</button>;
+    };
+    const { getByText } = render(
+      <AnalyticsFunnel funnelType="single-page" optionalStepNumbers={[]} totalFunnelSteps={3}>
+        <ChildComponent />
+      </AnalyticsFunnel>
+    );
+    act(() => void jest.runAllTimers());
+    fireEvent.click(getByText('Submit'));
+    await waitFor(() => {
+      expect(onCompleteSpy).toHaveBeenCalled();
+      removeCreateFlowDoneEvent(onCompleteSpy);
```

Comment:
This cleanup does not get executed when there's a problem with the test code itself or when the test fails. That's what `beforeEach` and `afterEach` are for. A more idiomatic approach would look something like this:

```
describe('emit cloudscape-creation-flow-done event', () => {
  const doneSpy = jest.fn();

  beforeEach(() => {
    document.addEventListener(CREATION_FLOW_DONE_EVENT_NAME, doneSpy);
    // ...
  });
  afterEach(() => {
    document.removeEventListener(CREATION_FLOW_DONE_EVENT_NAME, doneSpy);
  });
});
```